embedded/embedded-api/src/main/java/org/mule/runtime/module/embedded/internal/classloading/FilteringClassLoader.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.embedded.internal.classloading;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Boolean.valueOf;
+import static java.lang.String.format;
+import static java.lang.System.getProperty;
+import static java.util.Collections.emptyList;
+
+import org.mule.runtime.module.embedded.internal.NotExportedClassException;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Enumeration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO MULE-11882 - Consolidate classloading isolation
+public class FilteringClassLoader extends ClassLoader {
+
+  protected static final Logger logger = LoggerFactory.getLogger(FilteringClassLoader.class);
+  public static final String SYSTEM_PROPERTY_PREFIX = "mule.";
```

Comment:
It's not configurable and that's the constant.  This class already has a TODO to consolidate classloading logic in a separate, reusable module. Once that's that, this would exists only in that module.